src/cp/slipharden.cxx

```diff
@@ -937,6 +937,302 @@ void SimpleLinearHardening::consistency(Lattice & L) const
   }
 }
 
+LANLTiModel::LANLTiModel(ParameterSet & params):
+    SlipHardening(params),
+    tau_0_(params.get_object_parameter_vector<Interpolate>("tau_0")),
+    C_st_(params.get_object_parameter<SquareMatrix>("C_st")),
+    mu_(params.get_object_parameter_vector<Interpolate>("mu")),
+	  k1_(params.get_object_parameter_vector<Interpolate>("k1")), 
+    k2_(params.get_object_parameter_vector<Interpolate>("k2")),
+    X_s_(params.get_parameter<double>("X_s")),
+    varprefix_(params.get_parameter<std::string>("varprefix")), 
+    twinprefix_(params.get_parameter<std::string>("twinprefix"))
+{ 
+  // Shouldn't tau_0 be 24 and C_st be 12?
+  // Long term you should fix the /2
+  if (C_st_->n() != tau_0_.size()/2) {
+    throw std::invalid_argument("Twinning interaction matrix and initial strength sizes do not agree!");
+  }
+  
+  varnames_.resize(size());
+  for (size_t i = 0; i < size(); i++) {
+	if (i < size()/2){ // You'll need to fix this at some point
+      varnames_[i] = varprefix_+std::to_string(i);
+	} else{
+	  varnames_[i] = twinprefix_+std::to_string(i);	
+	}
+    
+  }
+  init_cache_();
+}
+
+std::string LANLTiModel::type()
+{
+  return "LANLTiModel";
+}
+
+std::unique_ptr<NEMLObject> LANLTiModel::initialize(ParameterSet & params)
+{
+  return neml::make_unique<LANLTiModel>(params);
+}
+
+ParameterSet LANLTiModel::parameters()
+{
+  ParameterSet pset(LANLTiModel::type());
+
+  pset.add_parameter<std::vector<NEMLObject>>("tau_0");
+  pset.add_parameter<NEMLObject>("C_st");
+  pset.add_parameter<std::vector<NEMLObject>>("mu");
+  pset.add_parameter<std::vector<NEMLObject>>("k1");
+  pset.add_parameter<std::vector<NEMLObject>>("k2");
+  pset.add_optional_parameter<double>("X_s", 0.9);
+  pset.add_optional_parameter<std::string>("varprefix", 
+                                           std::string("rho"));
+  pset.add_optional_parameter<std::string>("twinprefix", 
+                                           std::string("slip"));
+
+  return pset;
+}
+
+std::vector<std::string> LANLTiModel::varnames() const
+{
+  return varnames_;
+}
+
+void LANLTiModel::set_varnames(std::vector<std::string> vars)
+{
+  varnames_ = vars;
+  init_cache_();
+}
+
+void LANLTiModel::populate_history(History & history) const
+{
+  for (auto vn : varnames_) {
+    history.add<double>(vn);
+  }
+}
+
+void LANLTiModel::init_history(History & history) const
+{
+  size_t i = 0;
+  for (auto vn : varnames_) {
+    history.get<double>(vn) = 1.0e-6;
```

Comment:
Is this a "approximate" zero or a real initial value?

If it's a real initial value it should be a required parameter.

If it's a "approximate" zero just for numerical reasons it would be a parameter with a default value.

Generally we don't want "magic numbers" in the code like this.